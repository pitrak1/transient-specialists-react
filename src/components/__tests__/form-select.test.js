import React from 'react'
import FormSelect from '../form-select'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import { Select } from '@instructure/ui-forms'

describe('FormSelect', () => {
  const defaultOptionLabel = 'Select Something'
  let disabled
  const identifier = 'someIdentifier'
  const label = 'Some Label'
  let onChange
  const options = [
    { id: 1, name: 'option 1' },
    { id: 2, name: 'option 2' },
    { id: 3, name: 'option 3' },
  ]
  let required
  const value = null

  beforeEach(() => {
    disabled = false
    onChange = sinon.stub()
    required = false
  })

  const render = () =>
    shallow(
      <FormSelect
        defaultOptionLabel={defaultOptionLabel}
        disabled={disabled}
        identifier={identifier}
        label={label}
        onChange={onChange}
        options={options}
        required={required}
        value={value}
      />,
    )

  const getSelectProps = node => node.find(Select).props()

  it('renders a select with multiple options', () => {
    const node = render()
    expect(node).toMatchSnapshot()
  })

  it('is disabled if disabled is true', () => {
    disabled = true
    const node = render()
    expect(node).toMatchSnapshot()
  })

  it('calls onChange prop with identifier and numeric value when value is changed', () => {
    const node = render()
    getSelectProps(node).onChange(null, { value: '2' })
    expect(onChange.firstCall.args).toEqual([identifier, 2, expect.anything()])
  })

  it('sets messages if given default option when required', () => {
    required = true
    const node = render()
    getSelectProps(node).onChange(null, { value: '0' })
    expect(node).toMatchSnapshot()
  })

  it('calls onChange prop with false validity flag if given default value and required', () => {
    required = true
    const node = render()
    getSelectProps(node).onChange(null, { value: '0' })
    expect(onChange.firstCall.args[2]).toBe(false)
  })

  it('does not set messages if given default option when not required', () => {
    const node = render()
    getSelectProps(node).onChange(null, { value: '0' })
    expect(getSelectProps(node).messages).toEqual([])
    expect(node).toMatchSnapshot()
  })

  it('calls onChange prop with true validity flag if given default value and not required', () => {
    const node = render()
    getSelectProps(node).onChange(null, { value: '0' })
    expect(onChange.firstCall.args[2]).toBe(true)
  })
})
