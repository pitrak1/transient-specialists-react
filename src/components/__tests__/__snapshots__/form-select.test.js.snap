// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect does not set messages if given default option when not required 1`] = `
<div>
  <Select
    allowCustom={false}
    allowEmpty={true}
    closeOnSelect={true}
    constrain="window"
    disabled={false}
    editable={false}
    emptyOption="---"
    filter={[Function]}
    formatSelectedOption={[Function]}
    inputRef={[Function]}
    label="Some Label"
    loadingText={null}
    messages={Array []}
    multiple={false}
    onChange={[Function]}
    optionsMaxWidth={null}
    readOnly={false}
    selectedOption="0"
    size="medium"
    visibleOptionsCount={8}
  >
    <option
      key="0"
      value="0"
    >
      Select Something
    </option>
    <option
      key="1"
      value="1"
    >
      option 1
    </option>
    <option
      key="2"
      value="2"
    >
      option 2
    </option>
    <option
      key="3"
      value="3"
    >
      option 3
    </option>
  </Select>
  <p />
</div>
`;

exports[`FormSelect is disabled if disabled is true 1`] = `
<div>
  <Select
    allowCustom={false}
    allowEmpty={true}
    closeOnSelect={true}
    constrain="window"
    disabled={true}
    editable={false}
    emptyOption="---"
    filter={[Function]}
    formatSelectedOption={[Function]}
    inputRef={[Function]}
    label="Some Label"
    loadingText={null}
    messages={Array []}
    multiple={false}
    onChange={[Function]}
    optionsMaxWidth={null}
    readOnly={false}
    selectedOption="0"
    size="medium"
    visibleOptionsCount={8}
  >
    <option
      key="0"
      value="0"
    >
      Select Something
    </option>
    <option
      key="1"
      value="1"
    >
      option 1
    </option>
    <option
      key="2"
      value="2"
    >
      option 2
    </option>
    <option
      key="3"
      value="3"
    >
      option 3
    </option>
  </Select>
  <p />
</div>
`;

exports[`FormSelect renders a select with multiple options 1`] = `
<div>
  <Select
    allowCustom={false}
    allowEmpty={true}
    closeOnSelect={true}
    constrain="window"
    disabled={false}
    editable={false}
    emptyOption="---"
    filter={[Function]}
    formatSelectedOption={[Function]}
    inputRef={[Function]}
    label="Some Label"
    loadingText={null}
    messages={Array []}
    multiple={false}
    onChange={[Function]}
    optionsMaxWidth={null}
    readOnly={false}
    selectedOption="0"
    size="medium"
    visibleOptionsCount={8}
  >
    <option
      key="0"
      value="0"
    >
      Select Something
    </option>
    <option
      key="1"
      value="1"
    >
      option 1
    </option>
    <option
      key="2"
      value="2"
    >
      option 2
    </option>
    <option
      key="3"
      value="3"
    >
      option 3
    </option>
  </Select>
  <p />
</div>
`;

exports[`FormSelect sets messages if given default option when required 1`] = `
<div>
  <Select
    allowCustom={false}
    allowEmpty={true}
    closeOnSelect={true}
    constrain="window"
    disabled={false}
    editable={false}
    emptyOption="---"
    filter={[Function]}
    formatSelectedOption={[Function]}
    inputRef={[Function]}
    label="Some Label"
    loadingText={null}
    messages={
      Array [
        Object {
          "text": "Some Label is required",
          "type": "error",
        },
      ]
    }
    multiple={false}
    onChange={[Function]}
    optionsMaxWidth={null}
    readOnly={false}
    selectedOption="0"
    size="medium"
    visibleOptionsCount={8}
  >
    <option
      key="0"
      value="0"
    >
      Select Something
    </option>
    <option
      key="1"
      value="1"
    >
      option 1
    </option>
    <option
      key="2"
      value="2"
    >
      option 2
    </option>
    <option
      key="3"
      value="3"
    >
      option 3
    </option>
  </Select>
  <p />
</div>
`;
